integration_tests/steps/device.py

```diff
@@ -85,17 +89,20 @@ def step_impl(context):
     # placing bids and offers on every market cycle.
     # Should stop if an error occurs or if the simulation has finished
     counter = 0  # Wait for five minutes at most
-    while context.aggregator.errors == 0 and context.aggregator.status != "finished" and counter < 60:
+    while (
+            len(context.aggregator.errors) == 0
+            and context.aggregator.status != "finished"
+            and counter < 60):
         sleep(3)
         counter += 3
 
 
 @then("the external client does not report errors")
 def step_impl(context):
-    assert context.aggregator.errors == 0
+    assert len(context.aggregator.errors) == 0, \
+        f"The following errors were reported: {context.aggregator.errors}"
```

Comment:
This will show the actual error message, rather than just reporting the number of errors.